src/zinolib/event_types.py

```diff
@@ -0,0 +1,260 @@
+from datetime import datetime, timedelta, timezone
+from typing import Optional, ClassVar, List, TypeVar, Union, Dict, Generic
+
+from pydantic import ConfigDict, IPvAnyAddress
+from pydantic import BaseModel, computed_field
+
+from .compat import StrEnum
+
+
+def utcnow():
+    return datetime.now(timezone.utc)
+
+
+class AdmState(StrEnum):
+    """
+
+    Historic states:
+
+    * "active" was used twice in 1998
+    """
+    # ACTIVE = "active"
+
+    CLOSED = "closed"
+    CONFIRM_WAIT = "confirm-wait"
+    IGNORED = "ignored"
+    OPEN = "open"
+    WAITING = "waiting"
+    WORKING = "working"
+
+
+class FlapState(StrEnum):
+    FLAPPING = 'flapping'
+    STABLE = 'stable'
+
+
+class BFDState(StrEnum):
+    ADMIN_DOWN = "adminDown"
+    DOWN = "down"
+    INIT = "init"
+    UP = "up"
+
+
+class ReachabilityState(StrEnum):
+    REACHABLE = "reachable"
+    NORESPONSE = "no-response"
+
+
+class PortState(StrEnum):
+    """
+    Historic states:
+
+    * "testing" was used six times from 2000 to 2001
+    * "admin0" was used ten times from 2000 to 2002
+    * "flapping" was used 22 times on 2016-07-13T09:55 UTC
+    * "notPresent" was used 161 times during 2003
+    * "5" was used 389 times from 1998 to 2000
+    * "dormant" was used 813 times from 2000 to 2006
+    """
+    # TESTING = "testing"
+    # ADMIN_0 = "admin0"
+    # FLAPPING = "flapping"
+    # NOT_PRESENT = "notPresent"
+    # FIVE = "5"
+    # DORMANT = "dormant"
+
+    ADMIN_DOWN = "adminDown"
+    DOWN = "down"
+    LOWER_LAYER_DOWN = "lowerLayerDown"
+    UP = "up"
+
+
+class LogEntry(BaseModel):
+    date: datetime  # epoch
+    log: str
+
+    @classmethod
+    def create_list(cls, raw_log_list):
+        log_list = []
+        for entry in raw_log_list:
+            obj = cls(**entry)
+            log_list.append(obj)
+        return log_list
+
+
+class HistoryEntry(BaseModel):
+    log: str
+    date: datetime  # epoch
+    user: str
+
+    @classmethod
+    def create_list(cls, raw_history_list):
+        history_list = []
+        for entry in raw_history_list:
+            obj = cls(**entry)
+            history_list.append(obj)
+        return history_list
+
+
+class Event(BaseModel):
+    """
+    """
+    class Type(StrEnum):
+        ALARM = "alarm"
+        BFD = "bfd"
+        BGP = "bgp"
+        PORTSTATE = "portstate"
+        REACHABILITY = "reachability"
+
+    id: int
+    type: ClassVar[Type]
+    adm_state: AdmState
+    router: str
+    opened: datetime  # epoch
+
+    lastevent: Optional[str] = ''
+    lasttrans: Optional[datetime] = None  # epoch
+    updated: Optional[datetime] = None # epoch
+    polladdr: Optional[IPvAnyAddress] = None
+    priority: int = 100
+
+    # Cannot be set here, only per subclass
+    # port: str = ""
+    # description: str = ""
+    # op_state: str = ""
+
```

Comment:
The above is a rather important comment that will prevent a future refactor that *will* break things. It *could* be a doc-string instead of a comment though.